### ISSUE TITLE
Map Oracle type NUMBER(19,0) to long

### ISSUE THREAD

**Issue by marcushalonen** [confluentinc/kafka-connect-jdbc#393]
I have several Oracle tables with fields defined as NUMBER(19,0). I configured numeric.precision.mapping=true and was expecting long but I'm getting bytes.

On [DataConverter.java#L208](https://github.com/confluentinc/kafka-connect-jdbc/blob/master/src/main/java/io/confluent/connect/jdbc/source/DataConverter.java#L208) and [DataConverter.java#L399](https://github.com/confluentinc/kafka-connect-jdbc/blob/master/src/main/java/io/confluent/connect/jdbc/source/DataConverter.java#L399) there is a check for "precision < 19" Shouldn't this be "precision <= 19"?

In the proposal for https://github.com/confluentinc/kafka-connect-jdbc/issues/101 we had "precision < 20"

If I make the above changes I can ingest all my Oracle tables without any problems.

References:
- https://docs.oracle.com/cd/E19501-01/819-3659/gcmaz/
- https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/dialect/Oracle8iDialect.java#L143
- https://docs.oracle.com/cd/B10501_01/java.920/a96654/basic.htm#1021193

**Comment by szalapski** [comment]
This is a glaring issue that needs to be addressed for any Oracle numeric type  Not just NUMBER(19,0), but NUMBER(28,0), NUMBER(19,2), NUMBER(29,9), and of course, NUMBER with no specified precision

- No number type should serialize to a byte character string
- Higher precision/scale NUMBER(x) and NUMBER(x,y) types should work without need to CAST or use a string
- The "any precision" NUMBER type should work.

This is throwing us off everywhere.  Can you prioritize a fix for this?

**Comment by guija** [comment]
I completely agree to this. We're also stumbling over this every few weeks. Please prioritize this.

**Comment by jonariver** [comment]
@Confluent-KG you guys get paid. what are you waiting for? christmas in 2030?

**Comment by tkhasimbasha** [comment]
is this issue fixed..

**Comment by szalapski** [comment]
No.  They are ignoring it.

**Comment by kinghuang** [comment]
Very annoying that this still isn't addressed after all these years. My team work around this bug back in 2019, and still continues to do so. Confluent just isn't serious about Kafka Connect, it seems.

**Comment by YeonghyeonKO** [comment]
Notice me when you fix this issue. I'm eager to use Kafka Connect corretly
